File: modules/common/src/main/java/com/media/bus/common/web/response/ErrorView.java
package com.media.bus.common.web.response;

import com.fasterxml.jackson.annotation.JsonInclude;

import java.time.Instant;
import java.util.List;

/// API 에러 응답 래퍼.
///
/// `ExceptionAdvisor`가 단독으로 사용하는 순수 독립 레코드.
@JsonInclude(JsonInclude.Include.NON_NULL)
public record ErrorView(
		String code,
		String message,
		int status,
		String error,
		Instant timestamp,
		String path,
		List<FieldErrorDetail> errors
) {
	public record FieldErrorDetail(String field, String message) {}
}

File: modules/common/src/main/kotlin/com/media/bus/common/web/response/ErrorView.kt
package com.media.bus.common.web.response

import com.fasterxml.jackson.annotation.JsonInclude
import java.time.Instant

/**
 * ## API 에러 응답 래퍼
 *
 * `ExceptionAdvisor`가 단독으로 사용하는 순수 독립 데이터 클래스.
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
data class ErrorView(
    val code: String,
    val message: String,
    val status: Int,
    val error: String,
    val timestamp: Instant,
    val path: String,
    val errors: List<FieldErrorDetail>?,
) {
    data class FieldErrorDetail(val field: String, val message: String?)
}
